sdk/eventhub/eventhubs-checkpointstore-blob/src/storageBlobInterfaces.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  Metadata,
+  BlobItem,
+  ContainerListBlobFlatSegmentResponse,
+  ContainerListBlobsOptions,
+  BlockBlobUploadOptions,
+  BlockBlobUploadResponse,
+  HttpRequestBody,
+  BlobSetMetadataOptions,
+  ContainerSetMetadataResponse,
+} from "@azure/storage-blob";
+import { PagedAsyncIterableIterator } from "@azure/core-paging";
+
+/**
+ * An interface compatible with an instance of {@link BlobClient}.
+ */
+export interface BlobClientLike {
```

Comment:
Copying here: this also works...

```ts
import { ContainerClient, BlobClient, BlockBlobClient } from "@azure/storage-blob";

export type BlobClientLike = {
  [P in keyof Pick<BlobClient, "getBlockBlobClient">]: P extends "getBlockBlobClient"
    ? (...args: Parameters<BlobClient[P]>) => BlockBlobClientLike // getBlockBlobClient is mapped to return our interface
    : BlobClient[P];
};

export type ContainerClientLike = {
  [P in keyof Pick<ContainerClient, "getBlobClient" | "listBlobsFlat">]: P extends "getBlobClient"
    ? (...args: Parameters<ContainerClient[P]>) => BlobClientLike
    : ContainerClient[P];
};

export type BlockBlobClientLike = Pick<BlockBlobClient, "upload" | "setMetadata">;
```

I'm trying to decide which one to go with - any thoughts @jeremymeng @xirzec @witemple-msft

Reply:
honestly I like the simpler and more direct version rather than the calculated interface above

Reply:
I do too, thanks!